optimistix/_search.py

```diff
@@ -144,18 +156,48 @@ class ResidualJac(FunctionInfo, Generic[Y, Out], strict=True):
 
     residual: Out
     jac: lx.AbstractLinearOperator
-    grad: Y
-
-    def __init__(self, residual: Out, jac: lx.AbstractLinearOperator):
-        self.residual = residual
-        self.jac = jac
-        # The gradient is used ubiquitously, so compute it once here, so that it can be
-        # used without recomputation in both the descent and search.
-        self.grad = jac.transpose().mv(residual)
 
     def as_min(self):
         return 0.5 * sum_squares(self.residual)
 
+    def compute_grad(self):
+        # Not precomputed during `__init__` as this may hit reverse-mode autodiff which
+        # may not be valid.
+        if any(jnp.iscomplexobj(x) for x in jtu.tree_leaves(self.residual)):
+            conj_residual = jtu.tree_map(jnp.conj, self.residual)
+            conj_jac = lx.conj(self.jac)
+            return (
+                0.5
+                * (
+                    self.jac.transpose().mv(conj_residual) ** ω
+                    + conj_jac.transpose().mv(self.residual) ** ω
+                )
+            ).ω
```

Comment:
```suggestion
            conj_jac = lx.conj(self.jac)
            return (conj_jac.transpose().mv(self.residual) ** ω).ω
```
Something similar is needed in `grad_dot`. Maybe you want to pre-compute `conj_jac`. Note again that this is `df/dz*`, but that's the one you need for optimization. Alternatively, you can calculate `df/dz` and conjugate later, there is no difference for real-valued functions